rewrite-java/src/main/java/org/openrewrite/java/cleanup/NoDoubleBraceInitialization.java

```diff
@@ -207,4 +217,22 @@ public J.Block visitBlock(J.Block block, ExecutionContext ctx) {
             return bl;
         }
     }
+
+    private static class FindMethodInvocationInDoubleBrace extends JavaIsoVisitor<AtomicBoolean>{
+        /**
+         * Find whether any method invocation happened in the double brace.
+         * @param j The subtree to search, supposed to be the 2nd brace (J.Block)
+         * @return true if any method invocation found in the double brace, otherwise false.
+         */
+        static AtomicBoolean find(J j) {
+            return new FindMethodInvocationInDoubleBrace()
+                .reduce(j, new AtomicBoolean());
+        }
+
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, AtomicBoolean atomicBoolean) {
+            atomicBoolean.set(true);
+            return method;
+        }
+    }
```

Comment:
This doesn't look quite right. It would be totally fine for method invocations to be arguments to `add()` or `put()`. 
```java
List<String> a = new ArrayList<>() {{ add(foo()); }};
```

Reply:
I updated the logic of this method to support collection content initialization detection.
and this code will be turned into 
```java
List<String> a = new ArrayList<>();
a.add(foo());
```
and it's expected.